feature/gribi/mpls_in_udp/otg_tests/mpls_in_udp_scale/mpls_in_udp_scale_test.go

```diff
@@ -0,0 +1,1757 @@
+}
+
+// programMPLSinUDPEntries programs gRIBI entries for MPLS-in-UDP encapsulation.
+// It installs a single NextHop that performs MPLS-in-UDP encapsulation with the
+// provided outer IPv6 and UDP header attributes, associates multiple NextHopGroups
+// (NHGs) with that NextHop, and finally installs IPv6 /128 routes pointing to each NHG.
+func programMPLSinUDPEntries(t *testing.T, dut *ondatra.DUTDevice, mplsNHID, mplsLabelStart uint64, numNHGs int, outerIPv6Src, outerIPv6Dst string, outerDstUDPPort uint16, outerIPTTL, outerDSCP uint8) []fluent.GRIBIEntry {
+	t.Helper()
+	entries := []fluent.GRIBIEntry{
+		// Create the single MPLS-in-UDP NextHop
+		fluent.NextHopEntry().
+			WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(mplsNHID).
+			AddEncapHeader(
+				fluent.MPLSEncapHeader().WithLabels(mplsLabelStart),
+				fluent.UDPV6EncapHeader().
+					WithSrcIP(outerIPv6Src).
+					WithDstIP(outerIPv6Dst).
+					WithDstUDPPort(uint64(outerDstUDPPort)).
+					WithIPTTL(uint64(outerIPTTL)).
+					WithDSCP(uint64(outerDSCP)),
+			),
+	}
+	// Create NextHopGroups pointing to the same NH
+	for i := 0; i < numNHGs; i++ {
+		nhgID := mplsNHGID + uint64(i)
+		entries = append(entries,
+			fluent.NextHopGroupEntry().
+				WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+				WithID(nhgID).
+				AddNextHop(mplsNHID, 1),
+		)
+	}
+
+	// Create IPv6 routes pointing to those NHGs
+	baseIP := net.ParseIP(baseIPv6)
+	ipInt := big.NewInt(0).SetBytes(baseIP.To16())
+	for i := 0; i < numNHGs; i++ {
+		ipBytes := ipInt.FillBytes(make([]byte, 16))
+		prefix := fmt.Sprintf("%s/128", net.IP(ipBytes).String())
+		ipInt.Add(ipInt, big.NewInt(1))
+
+		nhgID := mplsNHGID + uint64(i)
+		entries = append(entries,
+			fluent.IPv6Entry().
+				WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+				WithPrefix(prefix).
+				WithNextHopGroup(nhgID).
+				WithNextHopGroupNetworkInstance(deviations.DefaultNetworkInstance(dut)),
+		)
+	}
+
+	return entries
+}
+
+func programMPLSinUDPMultiEntries(t *testing.T, dut *ondatra.DUTDevice, vrfs []string, mplsNHBase, mplsLabelStart uint64, numNHGs int, outerIPv6Src, outerIPv6Dst string, outerDstUDPPort uint16, outerIPTTL, outerDSCP uint8) []fluent.GRIBIEntry {
+	t.Helper()
+	var entries []fluent.GRIBIEntry
+
+	for vrfIdx, vrfName := range vrfs {
+		label := mplsLabelStart + uint64(vrfIdx)
+		nhID := mplsNHBase + uint64(vrfIdx)
+
+		// NextHop
+		entries = append(entries,
+			fluent.NextHopEntry().
+				WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+				WithIndex(nhID).
+				AddEncapHeader(
+					fluent.MPLSEncapHeader().WithLabels(label),
+					fluent.UDPV6EncapHeader().
+						WithSrcIP(outerIPv6Src).
+						WithDstIP(outerIPv6Dst).
+						WithDstUDPPort(uint64(outerDstUDPPort)).
+						WithIPTTL(uint64(outerIPTTL)).
+						WithDSCP(uint64(outerDSCP)),
+				),
+		)
+
+		// NextHopGroups and Routes
+		for i := 0; i < numNHGs; i++ {
```

Comment:
Since i is starting from 0, you can use the condensed format here.


```suggestion
		for i := range numNHGs {
```